volunteer-communication-center/functions/setup-flow.js

```diff
@@ -0,0 +1,145 @@
+  function deployStudio() {
+    return configureFlowForFlex()
+      .then(() => configureFlowWithAssistant())
+      .then(() => {
+        return client.studio.flows
+        .create({
+          commitMessage: 'Code Exchange automatic deploy',
+          friendlyName: 'Volunteer Communication Center',
+          status: 'published',
+          definition: flowDefinition,
+        })
+        .then((flow) => flow)
+        .catch((err) => { throw new Error(err.details) });
+      });
+  }
+
+
+  function getWorkspace() {
+    return client.taskrouter.workspaces
+      .list({limit: 20})
+      .then(workspaces => workspaces.find(workspace => workspace["friendlyName"] == "Flex Task Assignment"))
+      .catch((err) => { throw new Error(err.details) });
+  }
+
+
+  function fetchWorkFlow(ws) {
+    return client.taskrouter.workspaces(ws)
+      .workflows
+      .list({limit: 20})
+      .then(workflows => workflows.find(workflow => workflow["friendlyName"] == "Assign to Anyone"))
+      .catch((err) => { throw new Error(err.details) });
+  }
+
+  function getChannel(ws, type) {
+    return client.taskrouter.workspaces(ws)
+      .taskChannels
+      .list({limit: 20})
+      .then(taskChannels => taskChannels.find(channel => channel["uniqueName"] == type));
+  }
+
+  async function configureFlowForFlex() {
+
+    const workspace = await getWorkspace();
+    const workflow = await fetchWorkFlow(workspace.sid);
+    const smsChannel = await getChannel(workspace.sid, "chat");
+    const voiceChannel = await getChannel(workspace.sid, "voice");
+
+    if (workspace && workflow) {
+      const states = flowDefinition["states"];
+      for (let i = 0; i < states.length; i++) {
+        if (states[i]["properties"].hasOwnProperty("workflow")) {
+          
+          states[i]["properties"]["workflow"] = workflow.sid;
+
+          if (states[i].hasOwnProperty("name") && states[i]["name"] == "send_to_flex_sms") {
+            states[i]["properties"]["channel"] = smsChannel.sid;
+          } else if (states[i].hasOwnProperty("name") && states[i]["name"] == "send_to_flex_voice") {
+            states[i]["properties"]["channel"] = voiceChannel.sid;
+          }
+        }
+      }
+      flowDefinition["states"] = states;
+
+    }
+  }
+
+
+
+  function configureFlowWithAssistant() {
+    if (process.env.AUTOPILOT_SID) {
+      const states = flowDefinition["states"];
+
+      for (let i = 0; i < states.length; i++) {
+        if (states[i]["properties"].hasOwnProperty("autopilot_assistant_sid")) {
+          
+          states[i]["properties"]["autopilot_assistant_sid"] = process.env.AUTOPILOT_SID;
+        }
+      }
+      flowDefinition["states"] = states;
+
+    }
+  }
+
+
+  function getPhoneNumberSid() {
+    return new Promise((resolve, reject) => {
+      client.incomingPhoneNumbers
+        .list({ phoneNumber: context.TWILIO_PHONE_NUMBER, limit: 20 })
+        .then((incomingPhoneNumbers) => {
+          const n = incomingPhoneNumbers[0];
```

Comment:
you might need to check
```
if (incomingPhoneNumbers.length) {
  ...
}
```
since technically this could be an empty list, calling `incomingPhoneNumbers[0];` would generate a runtime error

Reply:
ah makes sense